src/main/java/runner/Parser.java

```diff
@@ -0,0 +1,187 @@
+package runner;
+import components.Deadline;
+import components.Event;
+import components.Task;
+import components.Todo;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parser to parse the command process the necessary actions.
+ */
+public class Parser {
+    private final Duke duke;
+
+    /**
+     * Constructor for Parser.
+     * @param duke a Duke chat-bot to work on.
+     */
+    public Parser(Duke duke) {
+        this.duke = duke;
+    }
+
+
+    /**
+     * @param info The information given by the user.
+     */
+    public void handle(String info) {
+        if (info.equals("bye")) {
+            terminate();
+        } else if (info.equals("list")) {
+            Ui.showList(duke.taskList);
+        } else {
+            read(info);
+        }
+    }
+
+    public void read(String info) {
+        if (!info.contains(" ")) {
+            System.out.println("Not Smart to Understand -_-");
+            return;
+        }
+        try {
+            String[] segments = info.split(" ", 2);
+            String first = segments[0];
+            String arg = segments[1];
+            switch (first) {
+                case "mark":
```

Comment:
In switch statements, there is no indentation for case clauses by the coding standard, so perhaps alter accordingly?

Reply:
Yes thanks!